### PR TITLE
scan pause when redeem queue full

### PR DESCRIPTION
**Pull request by xiaoch05**

### CODE REVIEW

**Comment by fewensa** [comment]
https://github.com/darwinia-network/bridger/blob/1373a4eaa3ed61d0b996eee505d4d6a25468f739/task/task-darwinia-ethereum/src/service/ethereum/ethereum_logs_handler.rs#L69

also modify darwinia-ethereum?

File: task/task-pangolin-ropsten/src/service/ropsten/ropsten_logs_handler.rs [hunk 1]
```diff
@@ -79,7 +79,12 @@ impl LogsHandler for RopstenLogsHandler {
         let check_position = if txs.is_empty() { to } else { from - 1 };
         self.check_redeem(check_position).await?;
         if self.waited_redeem.len() >= MAX_WAITED_REDEEM_COUNT {
-            // todo: there need stop scan running
+            info!(
+                target: PangolinRopstenTask::NAME,
+                "achieve redeem waited limit, pause scan, from: {:?}",
+                &from
+            );
+            self.tracker.flush_current(from as usize)?;
```
Comment (HackFisher) [inline task/task-pangolin-ropsten/src/service/ropsten/ropsten_logs_handler.rs:87]: Just a question, how does this pause the scan?

`flush_current` just update some key in the db

Comment (xiaoch05) [reply task/task-pangolin-ropsten/src/service/ropsten/ropsten_logs_handler.rs:87]: each time to scan, the tracker will read current from db during fast mode. So, if the current paused at `from` position, the tracker will always scan from this position.

**Comment by HackFisher** [comment]
> https://github.com/darwinia-network/bridger/blob/1373a4eaa3ed61d0b996eee505d4d6a25468f739/task/task-darwinia-ethereum/src/service/ethereum/ethereum_logs_handler.rs#L69
> 
> also modify darwinia-ethereum?

@xiaoch05 Any update on the darwinia-ethereum task?

**Review by fewensa — APPROVED** [review]

**Pull request merged**